clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1160,14 +1160,33 @@ static target getAccessTarget(QualType FieldTy,
 }
 
 bool SemaSYCL::isFreeFunction(const FunctionDecl *FD) {
-  for (auto *IRAttr : FD->specific_attrs<SYCLAddIRAttributesFunctionAttr>()) {
-    SmallVector<std::pair<std::string, std::string>, 4> NameValuePairs =
-        IRAttr->getAttributeNameValuePairs(getASTContext());
-    for (const auto &NameValuePair : NameValuePairs) {
-      if (NameValuePair.first == "sycl-nd-range-kernel" ||
-          NameValuePair.first == "sycl-single-task-kernel") {
+  clang::SourceLocation Loc = FD->getLocation();
```

Comment:
I'm pretty sure clang nns is not required here.
```suggestion
  SourceLocation Loc = FD->getLocation();
```